pilot/pkg/model/sidecar.go

```diff
@@ -280,20 +287,32 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
 		Namespace: sidecarConfig.Namespace,
 	}.HashCode())
 
-	egressConfigs := sidecar.Egress
+	if !features.EnableLazySidecarEvaluation {
+		initSidecarScope(ps, out, configNamespace)
+	} else {
+		out.initFunc = sync.OnceFunc(func() {
+			initSidecarScope(ps, out, configNamespace)
+		})
+	}
+
+	return out
+}
+
+func initSidecarScope(ps *PushContext, sidecarScope *SidecarScope, configNamespace string) {
```

Comment:
nit: there is a method PushContext.initSidecarScopes, so better rename to prevent misunderstanding

Reply:
this method is not under `PushContext`, I will rename it though